lib/experimental/fonts-library/class-wp-font-family.php

```diff
@@ -0,0 +1,421 @@
+
+		// Remove the uploaded font asset reference from the font face definition because it is no longer needed.
+		unset( $new_font_face['file'] );
+
+		// If the filepath has not a font mime type, we don't move the file and return the font face definition without src to be ignored later.
+		if ( ! WP_Font_Family_Utils::has_font_mime_type( $filepath ) ) {
+			return $new_font_face;
+		}
+
+		// Move the uploaded font asset from the temp folder to the wp fonts directory.
+		$file_was_moved = move_uploaded_file( $file['tmp_name'], $filepath );
+
+		if ( $file_was_moved ) {
+			// If the file was successfully moved, we update the font face definition to reference the new file location.
+			$new_font_face['src'] = WP_Fonts_Library::get_relative_fonts_path() . $filename;
+		}
+
+		return $new_font_face;
+	}
+
+	/**
+	 * Sanitizes the font family data using WP_Theme_JSON.
+	 *
+	 * @return array A sanitized font family defintion.
+	 */
+	private function sanitize() {
+		// Creates the structure of theme.json array with the new fonts.
+		$fonts_json = array(
+			'version'  => '2',
+			'settings' => array(
+				'typography' => array(
+					'fontFamilies' => array( $this->data ),
+				),
+			),
+		);
+		// Creates a new WP_Theme_JSON object with the new fonts to leverage sanitization and validation.
+		$theme_json     = new WP_Theme_JSON_Gutenberg( $fonts_json );
+		$theme_data     = $theme_json->get_data();
+		$sanitized_font = ! empty( $theme_data['settings']['typography']['fontFamilies'] )
+			? $theme_data['settings']['typography']['fontFamilies'][0]
+			: array();
+		$this->data     = $sanitized_font;
+		return $this->data;
+	}
+
+	/**
+	 * Downloads font face assets.
+	 *
+	 * Downloads the font face asset(s) associated with a font face. It works with both single
+	 * source URLs and arrays of multiple source URLs.
+	 *
+	 * @param array $font_face The font face array containing the 'src' attribute with the source URL(s) of the assets.
+	 * @return array The modified font face array with the new source URL(s) to the downloaded assets.
+	 */
+	private function download_font_face_assets( $font_face ) {
+		$new_font_face        = $font_face;
+		$srcs                 = ! empty( $font_face['src'] ) && is_array( $font_face['src'] )
+			? $font_face['src']
+			: array( $font_face['src'] );
+		$new_font_face['src'] = array();
+		$i                    = 0;
+		foreach ( $srcs as $src ) {
+			$filename = WP_Font_Family_Utils::get_filename_from_font_face( $font_face, $src, $i++ );
+			$new_src  = $this->download_asset( $src, $filename );
+			if ( $new_src ) {
+				$new_font_face['src'][] = $new_src;
+			}
+		}
+		if ( count( $new_font_face['src'] ) === 1 ) {
+			$new_font_face['src'] = $new_font_face['src'][0];
+		}
+		return $new_font_face;
+	}
+
+
+	/**
+	 * Downloads font face assets if the font family is a Google font, or moves them if it is a local font.
+	 *
+	 * @param array $files An array of files to be installed.
+	 * @return bool|WP_Error
```

Comment:
This method doesn't seem to return a `WP_Error` object. It only returns boolean values.
```suggestion
	 * @return bool
```